src/gitCommands.ts

```diff
@@ -0,0 +1,64 @@
+import { execSync } from 'node:child_process';
+import { existsSync, lstatSync } from 'node:fs';
+import { join } from 'node:path';
+
+export const isGitDirectory = (directoryPath: string): boolean => {
+	const gitPath = join(directoryPath, '.git');
+	return existsSync(gitPath) && lstatSync(gitPath).isDirectory();
+};
+
+export const isFileInGitDirectory = (filePath: string): boolean => {
+	try {
+		execSync(`git ls-files --error-unmatch ${filePath}`);
```

Comment:
is this command supported on all the git clients (e.g. Apple provides its own version of git)